### DISCUSSION TITLE
PS2 with two controllers connected, player 2 connection feels off, dropped or phantom inputs. Player 1 is always fine.

### DISCUSSION THREAD

**Discussion by klunkee** [darthcloud/BlueRetro#214] [Q&A]
I made an adapter with 2 PS2 cords and it works but there are issues with player 2. I need to test it more but I used several combinations of controllers to see if it was a conflict with a specific controller or not. I tried combinations of a Wii U Pro, PS4, and Wiimote with classic controller. It seemed like Player 1 was always stable no matter which controller combination was used but player 2 felt sluggish with odd input behavior every time.

**Comment by darthcloud** [comment]
For multiplayer, best result often come from using same controller type together.

WiiU Pro is probably the worst when used with other controller. See issue #3 .

I haven't notice any issue mixing wiimote and PS4 however.

**Reply by klunkee** [reply]
I will try testing with two controllers of the same type and see if it occurs again.

I'll also test the Wiimote PS4 combination again but I am pretty sure it was happening with that. Do you know if the ESP32 is prone to interference when the controllers are physically close to one another?

**Reply by klunkee** [reply]
I tried testing again and for some reason the controllers would pair but the ps2 was not receiving any of the inputs. I reflashed the ESP32 with the universal firmware and it finally started working. It looks like it somehow also fixed the laggy input bug and it worked fine when I tested it this time even with the Wii U Pro as player 1. Are there any major differences between the universal firmware and the ones for each console?

**Reply by darthcloud** [reply]
The only diff is at boot universal try to identify the system, and then start right mode, specific one start right away in the hardcoded console mode.
